map/mapObject.mjs

```diff
@@ -0,0 +1,102 @@
+"use strict"
+export {MapObject, MapIcon}
+
+import {worldSpaceToMapSpace, mapSpaceToScreenSpace, iconH} from "../map.js"
+import {Point} from "./point.mjs"
+
+/**
+ * An object that will be displayed on the map canvas.
+ */
+function MapObject(){
+    this.minX = 0;
+    this.maxX = 0;
+    this.minY = 0;
+    this.maxY = 0;
```

Comment:
why do we store maxX and maxY instead of width and height?
I originally did this because bounding box calclations are easier with maxX and maxY. but the drawing APIs all take width and height. So we need both of them. I'm still unsure which one to use, so I just didn't change it after I decided the first time. If you think it would be beneficial to switch, we can do that.

Reply:
why not both?

or we could make a function to give a shorthand reference without having to write out maxX - minX, etc.